R/statistics.R

```diff
@@ -691,3 +691,247 @@ SecretarySampler <- R6::R6Class("SecretarySampler", public = list(
     }
 )
 )
+#' Create an abstract streamer for the multi-armed bandit problem.
+#'
+#' @description \code{MultiArmedBandit}
+#'
+#' @docType class
+#'
+#'
+#' @export
+#' @format An \code{\link{R6Class}} generator object
+MultiArmedBandit <- R6::R6Class(
+    "MultiArmedBandit",
+    public = list(
+        #' @description Creates a new \code{MultiArmedBandit} streamer object.
+        #'
+        #' @param K the number of arms
+        #' @param T number of rounds / draws
+        #'
+        #' @return The new \code{MultiArmedBandit} (invisibly)
+        initialize = function(K, T, N = NULL) {
+            private$K <- K
+            private$T <- T
+            arm_info <- list()
+            for (k in seq(K)) {
+                arm_info[[k]] <- list(mean = CMA$new())
+            }
+            private$arm_info <- arm_info
+            invisible(self)
+        },
+        #' @description Update the \code{MultiArmedBandit} streamer object.
+        #'
+        #' @param x a single draw from an arm
+        #' @param k index of the chosen arm
+        #'
+        #'
+        #' @return The updated code{MultiArmedBandit} (invisibly)
+        update = function(x, k) {
```

Comment:
If `k` comes from the previous `value`, would it be cleaner to remember that internally so that `k` can be `NULL` be default but specified if the user ignore the streamer's suggestion?

Reply:
I like this suggestion.